prov/rxm/src/rxm_domain.c

```diff
@@ -262,6 +325,14 @@ int rxm_domain_open(struct fid_fabric *fabric, struct fi_info *info,
 		goto err3;
 	}
 
+	/* We maintain an RMA key to MR map used for emulated atomic access
+	 * and bounds validation. We turn off the map mode bit FI_MR_PROV_KEY
+	 * since we specify the key used by MSG_EP provider. */
+	rxm_domain->util_domain.mr_map.mode &= ~FI_MR_PROV_KEY;
```

Comment:
Can we do that? The user then might expect that they could dictate the MR key -- which isn't true for verbs if I recall correctly.

Reply:
This change does not alter the actual MR mode bits of the RxM or Core provider. It is odd, but is required because the MR map common code normally acts as if it is creating software defined MRs. Since we are only using it to store the mappings of hardware generated MRs we need to supply the Keys. By setting this value, the common code will use the keys we pass and not try to update the MR attributes with a new key it generates.